integration_tests/gdk/components/test_integ_BuildCommand.py

```diff
@@ -0,0 +1,491 @@
+        mock_file.assert_any_call(file_name, "w")
+        mock_yaml_dump.call_count == 0
+    assert mock_get_proj_config.assert_called_once
+    mock_subprocess_run.assert_called_with(["gradle", "build"])  # called gradle build command
+    assert mock_copy_dir.call_count == 0  # No copying directories
+    assert supported_build_system.call_count == 1
+    assert mock_archive_dir.call_count == 0  # Archvie never called in gralde
+    assert mock_boto3_client.call_count == 0  # artifacts found in s3
+    assert mock_clean_dir.call_count == 1  # clean greengrass-build
+    assert mock_create_dir.call_count == 2  # create gg directories
+    assert mock_copy_file.call_count == 1
+    assert mock_exists.called
+
+
+def test_build_run_default_gradle_yaml_error_creating_recipe(mocker, supported_build_system, rglob_build_file):
+
+    mock_clean_dir = mocker.patch("gdk.common.utils.clean_dir", return_value=None)
+    mock_create_dir = mocker.patch("pathlib.Path.mkdir", return_value=None)
+    mock_copy_dir = mocker.patch("shutil.copytree", return_value=None)
+    mock_archive_dir = mocker.patch("shutil.make_archive", return_value=None)
+    pc = project_config()
+    pc["component_build_config"] = {"build_system": "gradle"}
+    pc["component_recipe_file"] = Path("/src/GDK-CLI-Internal/tests/gdk/static/build_command/recipe.yaml")
+    mock_get_proj_config = mocker.patch(
+        "gdk.commands.component.project_utils.get_project_config_values",
+        return_value=pc,
+    )
+
+    mock_boto3_client = mocker.patch("boto3.client")
+    mock_subprocess_run = mocker.patch("subprocess.run")
+    mock_yaml_dump = mocker.patch("yaml.dump", side_effect=Exception("writing failed"))
+    pc = mock_get_proj_config.return_value
+    mock_is_artifact_in_build = mocker.patch.object(BuildCommand, "is_artifact_in_build", return_value=True)
+    file_name = Path(pc["gg_build_recipes_dir"]).joinpath(pc["component_recipe_file"].name).resolve()
+    with patch("builtins.open", mock_open()) as mock_file:
+        with pytest.raises(Exception) as e:
+            parse_args_actions.run_command(CLIParser.cli_parser.parse_args(["component", "build"]))
+            mock_file.assert_any_call(file_name, "w")
+            mock_yaml_dump.call_count == 1
+        assert "Failed to create build recipe file at" in e.value.args[0]
+    assert mock_get_proj_config.assert_called_once
+    mock_subprocess_run.assert_called_with(["gradle", "build"])  # called gradle build command
+    assert mock_copy_dir.call_count == 0  # No copying directories
+    assert supported_build_system.call_count == 1
+    assert mock_is_artifact_in_build.call_count == 1
+    assert mock_archive_dir.call_count == 0  # Archvie never called in gralde
+    assert mock_boto3_client.call_count == 0  # artifacts found in s3
+    assert mock_clean_dir.call_count == 1  # clean greengrass-build
+    assert mock_create_dir.call_count == 2  # create gg directories
+
+
+def project_config():
```

Comment:
can this be put into testData as a config file and loaded in from there instead?